src/Kiota.Builder/Writers/Shell/ShellCodeMethodWriter.cs

```diff
@@ -415,84 +542,91 @@ private static void WriteCommandDescription(CodeMethod codeElement, LanguageWrit
         return builder;
     }
 
-    private void WriteContainerCommand(CodeMethod codeElement, LanguageWriter writer, CodeClass parent, string name)
+    private void WriteNavCommand(CodeMethod codeElement, LanguageWriter writer, CodeClass parent, string name)
     {
-        writer.WriteLine($"var command = new Command(\"{name}\");");
-        WriteCommandDescription(codeElement, writer);
-
-        if ((codeElement.AccessedProperty?.Type) is CodeType codeReturnType)
+        if (((codeElement.AccessedProperty?.Type) is CodeType codeReturnType) && (codeReturnType.TypeDefinition is CodeClass typeDef))
         {
             var targetClass = conventions.GetTypeString(codeReturnType, codeElement);
 
-            var builderMethods = codeReturnType.TypeDefinition?.GetChildElements(true).OfType<CodeMethod>()
+            var builderMethods = typeDef.Methods // Already ordered by name
                 .Where(static m => m.IsOfKind(CodeMethodKind.CommandBuilder))
-                .OrderBy(static m => m.Name)
-                .ThenBy(static m => m.ReturnType.IsCollection) ??
+                .OrderBy(static m => m.ReturnType.IsCollection)

```

Comment:
```suggestion
```

no need to re-order here

Reply:
I wanted the collections to be evaluated first. The default order is only by name